CONTRIBUTING.md

```diff
@@ -1,20 +1,69 @@
-# Contribution Guide
+# Contributing to AdoptOpenJDK/infrastructure
 
-The infrastructure project contains:
+Thank you for your interest in AdoptOpenJDK/infrastructure!
+
+We welcome and encourage all kinds of contributions to the project, not only
+code. This includes bug reports, user experience feedback, assistance in
+reproducing issues and more.
+
+The AdoptOpenJDK/infrastructure project contains:
 
 1. The [Ansible Playbooks](ansible/playbooks) for bootstrapping the build and test hosts (including a way to test Ansible)
 1. The overriding [Documentation](docs) for the build farm
 
-To make a change please:
+
+## Submitting a contribution to AdoptOpenJDK/infrastructure9
+
+You can propose contributions by sending pull requests (PRs) through GitHub.
+Following these guidelines will help us merge your pull requests smoothly:
+
+1. Your pull request is an opportunity to explain both what changes you'd like
+   pulled in, but also _why_ you'd like them added. Providing clarity on why
+   you want changes makes it easier to accept, and provides valuable context to
+   review.
+
+2. Follow the commit guidelines found below.
+
+3. We encourage you to open a pull request early, and mark it as "Work In
+   Progress", by prefixing the PR title with "WIP". This allows feedback to
+   start early, and helps create a better end product. Committers will wait
+   until after you've removed the WIP prefix to merge your changes.
+
+
+## Commit Guidelines
+
+The first line describes the change made. It is written in the imperative mood,
+and should say what happens when the patch is applied. Keep it short and
+simple. The first line should be less than 70 characters, where reasonable,
+and should be written in sentence case preferably not ending in a period.
+Leave a blank line between the first line and the message body.
+
+The body should be wrapped at 72 characters, where reasonable.
+
+Include as much information in your commit as possible. You may want to include
+designs and rationale, examples and code, or issues and next steps. Prefer
+copying resources into the body of the commit over providing external links.
+Structure large commit messages with headers, references etc. Remember, however,
+that the commit message is always going to be rendered in plain text.
+
+When a commit has related issues or commits, explain the relation in the message
+body. When appropriate, use the keywords described in the following help article
+to automatically close issues.
+https://help.github.com/articles/closing-issues-using-keywords/
+For example:
+
+```
+Install OpenSSL in windows playbook
+
+OpenSSL is required to compile java on windows, so the OpenSSL role will
+ensure the 32bit and 64bit versions are both installed into C:\openjdk
+
+Fixes: #1234
+```
+
+All changes should be made to a personal fork of AdoptOpenJDK/infrastructure and tested via Vagrant as detailed below.
```

Comment:
I'm not sure about this in the general case since if someone is modifying the AIX playbook (or even something specific to macos) then mandating this seems a touch overkill (even though it will verify that we don't break it elsewhere...)

Reply:
Agreed, may be tested in Vagrant where possible and on a 'test' system (To be Provided by Adopt at some stage in the future)

Reply:
I removed the last part of the sentence and left the testing section at the bottom with the TODO.